lib/brewdler/dsl.rb

```diff
@@ -1,23 +1,47 @@
 module Brewdler
   class Dsl
+    class Entry
+      attr_reader :type, :name, :extra
+
+      def initialize(type, name, extra={})
+        @type = type
+        @name = name
+        @extra = extra
+      end
+    end
+
+    attr_reader :entries
+
     def initialize(input)
       @input = input
+      @entries = []
     end
 
     def process
       instance_eval(@input)
+      self
+    end
+
+    def install
+      @entries.each do |entry|
+        case entry.type
+        when :brew then Brewdler::BrewInstaller.install(entry.name, entry.extra[:options])
+        when :cask then Brewdler::CaskInstaller.install(entry.name)
+        when :repo then Brewdler::RepoInstaller.install(entry.name)
+        end
+      end
     end
 
     def brew(name, options={})
-      Brewdler::BrewInstaller.install(name, options)
+      @entries << Entry.new(:brew, name, {:options => options})
```

Comment:
I think that's probably overkill, personally. If a single hash is enough now let's just use that and we can always nest them later if we need to (there won't be external API consumers so should be easy to do).